docs/architecture/adr-001-poap-contract.md

```diff
@@ -0,0 +1,177 @@
+
+```rust
+pub struct InstantiateMsg {
+pub cw721_code_id: u64,
+pub cw721_instantiate_msg: Cw721InstantiateMsg,
+pub event_info: EventInfo,
+}
+```
+
+* The `cw721_code_id` refers to a previously uploaded `CW721-base` contract on the chain;
+* The `cw721_instantiate_msg` contains the info to instantiate the `CW721-base`;
+* The `event_info` contains the event info.
+
+#### Cw721InstantiateMsg
+The following message instantiate the [CW721-base contract](https://github.com/CosmWasm/cw-nfts/tree/main/contracts/cw721-base):
+```rust
+pub struct Cw721InstatiateMsg {
+  pub name: String,
+  pub symbol: String,
+  pub minter: String,
+}
+```
+
+In the `POAP` contract case:
+* The `name` identifies the event name;
+* The `symbol` identifies the event logo (if exists);
+* The `minter` identifies the `POAP` contract address.
+
+##### EventInfo
+The `EventInfo` are used to instantiate the contract state with the information of the event.
+
+```rust
+pub struct EventInfo {
+    pub creator: Addr,
+    pub admin: Addr,
+    pub start_time: Timestamp,
+    pub end_time: Timestamp,
+    pub per_address_limit: u32,
+    pub base_poap_uri: String,
+    pub event_uri: String,
+    pub cw721_code_id: u64,
+}
+```
+
+* The `creator` field identifies the address of the event's creator;
+* The `admin` field identifies an optional admin for the contract, if not specified, the creator is the admin;
+* The `start_time` identifies the start time of the event;
+* The `end_time` identifies the end time of the event;
+* The `per_address_limit` identifies the max num of tokens that can be minted by an address;
+* The `base_poap_uri` identifies a valid `IPFS` URI corresponding to where the assets and metadata of the POAPs are stored.
+  * The `metadata` file is a `.json` that follow the `ERC-721` [metadata standard](https://docs.opensea.io/docs/metadata-standards#metadata-structure)
+* The field will be used to initialize the `token_uri` field of the `cw721-base` [`MintMsg<T>`](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61)
+* The `event_uri` field is used to initialize the `extension` part of the `cw721-base` [MintMsg< T >](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61) and contains a valid `IPFS` URI corresponding to where a `.json` file with the following event's metadata is stored:
+
+```json
+{
+  "name": "My awesome event",
+  "description": "Brief description of the event",
+  "city": "city where the event will be taking place",
+  "country": "country where the event will be taking place",
+  "start_date": "Start date for your event",
+  "end_date": "End date for your event",
+  "expiry_date": "From this day onward, no attendees will be able to mint POAPs from your event",
+  "year": <number>,
+  "event_url": "Event URL",
+  "virtual_event": true or false,
+}
+```
+
+#### Execute
+```rust
+pub enum ExecuteMsg {
```

Comment:
As I know, NFT ERC721 standard has a set of methods related to trade, like `transfer`, `approve` and etc. 
It would be better to note that NFTs in the contract is not tradable.

Reply:
From my side, Instead of transfer the POAP token to another wallet, connecting new wallet to the profile is the best choice by `x/profiles`. Disallowing transfer would brings more reliability to the token used as certificate.
About the benefit, it depends on what the event host brings the value off chain. For instance, the users having the POAP token can get a chance to get free coffee per day, then user would like to sell or borrow their POAP tokens.

Reply:
I got your point but I don't think we should see POAP as certificates. POAP proves the attendance to a particular event, so ideally even if tradable users could and could not create a secondary market to trade them, it depends on the value they give them. If the POAP guarantees some benefits, and they sell it, they also sell the benefits to another user, loosing them forever. I think we should give them the possibility to freely trade them, even if we will note give them a concrete UI to do so. Maybe one day someone will and there will be a complete new market for POAPs.
It's also possible for events creator to check on user POAP original minter and if it's different than the owner, suspend the benefits that particular POAP give.

Reply:
Can we have an another flag to swtich the status of tradable? Then, the contract can be used in both cases.

Reply:
Yeah, you are right. Under the current structure(3 contracts), tradable or not is controlled by cw721.